watertap/examples/flowsheets/RO_with_energy_recovery/RO_with_energy_recovery.py

```diff
@@ -101,41 +106,79 @@ def build():
             "concentration_polarization_type": ConcentrationPolarizationType.calculated,
         }
     )
-    m.fs.product = Product(default={"property_package": m.fs.properties})
-    m.fs.disposal = Product(default={"property_package": m.fs.properties})
-
-    # costing
-    m.fs.P1.costing = UnitModelCostingBlock(
-        default={"flowsheet_costing_block": m.fs.costing}
-    )
-    m.fs.P2.costing = UnitModelCostingBlock(
-        default={"flowsheet_costing_block": m.fs.costing}
-    )
-
     m.fs.RO.costing = UnitModelCostingBlock(
         default={"flowsheet_costing_block": m.fs.costing}
     )
-    m.fs.PXR.costing = UnitModelCostingBlock(
-        default={"flowsheet_costing_block": m.fs.costing}
-    )
+
+    # --- ERD blocks ---
+    if erd_type == ERDtype.pressure_exchanger:
+        m.fs.S1 = Separator(
+            default={"property_package": m.fs.properties, "outlet_list": ["P1", "PXR"]}
+        )
+
+        m.fs.PXR = PressureExchanger(default={"property_package": m.fs.properties})
+        m.fs.P2 = Pump(default={"property_package": m.fs.properties})
+        m.fs.M1 = Mixer(
+            default={
+                "property_package": m.fs.properties,
+                "momentum_mixing_type": MomentumMixingType.equality,  # booster pump will match pressure
+                "inlet_list": ["P1", "P2"],
+            }
+        )
+
+        # add costing for PX and recirculation pump
+        m.fs.PXR.costing = UnitModelCostingBlock(
+            default={"flowsheet_costing_block": m.fs.costing}
+        )
+        m.fs.P2.costing = UnitModelCostingBlock(
+            default={"flowsheet_costing_block": m.fs.costing}
+        )
+        # mixer and separator have no associated costing
+    elif erd_type == ERDtype.pump_as_turbine:
+        # add energy recovery turbine block
+        m.fs.ERD = EnergyRecoveryDevice(default={"property_package": m.fs.properties})
+        # add costing for ERD config
+        m.fs.ERD.costing = UnitModelCostingBlock(
+            default={"flowsheet_costing_block": m.fs.costing}
+        )
+    else:
+        raise NotImplementedError(
+            "erd_type was {}, but can only "
+            "be pressure_exchanger or pump_as_turbine"
+            "".format(erd_type)
+        )
```

Comment:
So, I half wonder if we move this to a function and call that function here and everywhere else you have a `pass`.

If somebody makes further modifications to the flowsheet, that would be safer.

Reply:
Added a function and test for this.